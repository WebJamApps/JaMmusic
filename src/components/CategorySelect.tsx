import {
  FormControl, InputLabel, MenuItem, Select,
} from '@mui/material';
import { Isong } from 'src/providers/Data.provider';

export function CategorySelect({ songJson, setFunc }:
{ songJson: Isong, setFunc: (arg0: Isong) => void }) {
  return (
    <FormControl fullWidth sx={{ marginBottom: '12px' }}>
      <InputLabel id="select-category-label">Category</InputLabel>
      <Select
        style={{ marginBottom: '12px' }}
        labelId="select-category-label"
        id="select-category"
        value={songJson.category}
        label="Category"
        onChange={(evt) => {
          const { target: { value } } = evt;
          setFunc({ ...songJson, category: value });
          return value;
        }}
      >
        <MenuItem value="original">original</MenuItem>
        <MenuItem value="mission">mission</MenuItem>
        <MenuItem value="pub">pub</MenuItem>
      </Select>
    </FormControl>
  );
}
