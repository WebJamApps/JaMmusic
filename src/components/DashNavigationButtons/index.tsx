import React from 'react';
import type { MusicDashboard } from '../../containers/MusicDashboard';

type PageProps = { comp:MusicDashboard };

const DashButtons = ({ comp }:PageProps): JSX.Element => (
  <div
    className="Nav-Buttons"
    id="Nav-Buttons"
    style={{
      display: 'inline',
    }}
  >
    <button
      className="floatRight1"
      type="button"
      id="Photos-Button"
      style={{ margin: '5px', fontSize: '55%' }}
      onClick={comp.handleNavClick}
    >
      Photos
    </button>
    <button
      className="floatRight2"
      type="button"
      id="Tours-Button"
      style={{ margin: '5px', fontSize: '55%' }}
      onClick={comp.handleNavClick}
    >
      Tours
    </button>
    <button
      className="floatRight3"
      type="button"
      id="Songs-Button"
      style={{ margin: '5px', fontSize: '55%' }}
      onClick={comp.handleNavClick}
    >
      Songs
    </button>
  </div>
);

export const DashNavigationButtons = ({ comp }:PageProps): JSX.Element => (
  <nav className="white">
    <div
      className="center"
      style={{
        textAlign: 'center',
      }}
    >
      <h4
        className="Musicdash-Title"
        style={{
          margin: '14px', fontWeight: 'bold', color: 'black',
        }}
      >
        Music Dashboard
        {DashButtons({ comp })}
      </h4>
    </div>
  </nav>
);
export default DashNavigationButtons;
