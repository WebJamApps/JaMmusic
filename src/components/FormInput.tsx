import { TextField } from '@mui/material';

interface IformInputProps {
  type: string | undefined,
  label: string, isRequired: boolean | undefined,
  onChange: ((event: React.ChangeEvent<HTMLInputElement>) => void) | undefined,
  value: string, style?: any
}
export const FormInput = (props:IformInputProps): JSX.Element => {
  const { label, style, isRequired, type, onChange, value } = props;
  let fId = label.toLowerCase();
  fId = fId.replace(/\s/g, '');
  const fIdArr = fId.split('(');
  // eslint-disable-next-line prefer-destructuring
  fId = fIdArr[0];
  return (
        <TextField
            style={style}
            id={fId}
            placeholder={isRequired ? '* '.concat(label) : label}
            type={type}
            name={fId}
            onChange={onChange}
            value={value || ''}
        />
  );
};
