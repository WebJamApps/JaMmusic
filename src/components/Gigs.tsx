import React, { useContext } from 'react';
import { DataGrid, GridColumns, GridEnrichedColDef, GridRenderCellParams } from '@mui/x-data-grid';
import { GigsContext } from '../providers/Gigs.provider';
import HtmlReactParser from 'html-react-parser';
import './Gigs.scss';

export const makeVenue = ():GridEnrichedColDef => {
  return (
    {
      field: 'venue',
      headerName: 'Venue',
      width: 600,
      editable: false,
      renderCell:(params:GridRenderCellParams) => <span>{HtmlReactParser(params.value)}</span>,
    }
  );
};

const columns:GridColumns = [
  {
    field: 'date',
    headerName: 'Date',
    width: 150,
    editable: false,
  },
  {
    field: 'time',
    headerName: 'Time',
    width: 150,
    editable: false,
  },
  {
    field: 'location',
    headerName: 'Location',
    width: 150,
    editable: false,
  },
  makeVenue(),
  {
    field: 'tickets',
    headerName: 'Tickets',
    width: 140,
    editable: false,
  },
];

export const Gigs = (): JSX.Element => {
  const { gigs } = useContext(GigsContext);
  return (
    <div style={{ margin: 'auto', padding:'10px', width:'100%' }}>
      <h4 style={{ textAlign:'center' }}>Gigs</h4>
      <div style={{ height: 400, width: '100%' }}>
      <DataGrid
        rows={gigs}
        columns={columns}
        pageSize={5}
        rowsPerPageOptions={[5]}
        disableSelectionOnClick
      />
      </div>
    </div>
  );
};
