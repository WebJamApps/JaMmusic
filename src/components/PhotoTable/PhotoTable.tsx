import React, { Dispatch } from 'react';
import MUIDataTable, { MUIDataTableColumnDef } from 'mui-datatables';
import ReactHtmlParser from 'react-html-parser';
import type { AGClientSocket } from 'socketcluster-client';
import 'core-js/stable';
import { connect } from 'react-redux';
import superagent from 'superagent';
import type { AnyAction } from 'redux';
import mapStoreToProps, { Iimage } from '../../redux/mapStoreToProps';

type Pprops = {
  dispatch: Dispatch<AnyAction>,
  picUpdated?: boolean,
  auth: { token: string },
  images: Iimage[],
  scc?: AGClientSocket
};
interface Pstate {
  columns: MUIDataTableColumnDef[]
}
export class PhotoTable extends React.Component<Pprops, Pstate> {
  superagent: superagent.SuperAgentStatic;

  constructor(props: Readonly<Pprops>) {
    super(props);
    this.superagent = superagent;
    this.setColumns = this.setColumns.bind(this);
    this.handleHideTable = this.handleHideTable.bind(this);
    this.addThumbs = this.addThumbs.bind(this);
    this.state = {
      columns: [],
    };
  }

  componentDidMount(): void { this.setColumns(); }

  handleHideTable(): boolean {
    const { dispatch } = this.props;
    dispatch({ type: 'SHOW_TABLE', showTable: false });
    return true;
  }

  setColumns(): void {
    const columns: MUIDataTableColumnDef[] = [];
    const titles = ['Thumbnail', 'Title', 'Caption', 'Link', 'Type', 'Modify'];
    for (let i = 0; i < titles.length; i += 1) {
      const label = titles[i];// eslint-disable-line security/detect-object-injection
      columns.push({
        name: label.toLowerCase(),
        label,
        options: {
          filter: false,
          sort: true,
          customBodyRender: (value: string) => (
            <div style={{
              margin: 0, fontSize: '12pt', maxWidth: '200px',
            }}
            >
              {label !== 'Modify' ? ReactHtmlParser(value) : value}
            </div>
          ),
        },
      });
    }
    this.setState({ columns });
  }

  /* let res: superagent.Response;
      try {
        res = await this.superagent.delete(`${process.env.BackendUrl}/book/${id}`)
          .set('Authorization', `Bearer ${auth.token}`).set('Accept', 'application/json');
      } catch (e) { return `${e.message}`; }
      if (res.status === 200) { window.location.reload(); return 'deleted pic'; }
      return `${res.status} ${res.body}`;
    }
    return 'no delete'; */

  deletePic(id: string | undefined): boolean { // eslint-disable-next-line no-restricted-globals
    const result = confirm('Deleting picture, are you sure?');// eslint-disable-line no-alert
    if (result) {
      const { scc, auth } = this.props;
      const pic = { id };
      if (scc && auth) {
        scc.transmit('deleteImage', { pic, token: auth.token });
        window.location.assign('/music');
        return true;
      } return false;
    } return false;
  }

  editPic(editImage: Iimage): boolean {
    const { dispatch } = this.props;
    // eslint-disable-next-line no-param-reassign
    // delete picData.modify;
    dispatch({ type: 'EDIT_PIC', editImage });
    return true;
  }

  addThumbs(arr: Iimage[]): Iimage[] {
    const newArr = arr;/* eslint-disable security/detect-object-injection */
    for (let i = 0; i < arr.length; i += 1) { // eslint-disable-next-line security/detect-object-injection
      newArr[i].thumbnail = `<img src=${arr[i].url} width="200px"/>`;
      const deletePicId = `deletePic${newArr[i]._id}`;// eslint-disable-line security/detect-object-injection
      const editPicId = `editPic${newArr[i]._id}`;// eslint-disable-line security/detect-object-injection
      newArr[i].link = `<a href=${arr[i].url} target="_blank">click to view</a>`;// eslint-disable-line security/detect-object-injection
      newArr[i].caption = newArr[i].comments === 'showCaption' ? 'display' : 'hide';
      newArr[i].modify = (// eslint-disable-line security/detect-object-injection
        <div>
          <button type="button" id={deletePicId} onClick={() => this.deletePic(newArr[i]._id)}>Delete Pic</button>
          <p>{' '}</p>
          <button type="button" id={editPicId} onClick={() => { this.editPic(newArr[i]); this.handleHideTable(); }}>
            Edit Pic
          </button>
        </div>
      );
    }
    return newArr;
  }

  render(): JSX.Element {
    const { columns } = this.state;
    const {
      images,
    } = this.props;
    let arr: Iimage[] = images;
    arr = this.addThumbs(arr);
    return (
      <div className="photoTable">
        <div style={{ maxWidth: '9in', margin: 'auto' }}>
          <MUIDataTable
            options={{
              filterType: 'dropdown',
              pagination: false,
              responsive: 'standard',
              filter: false,
              download: false,
              search: false,
              print: false,
              viewColumns: false,
              selectableRows: 'none',
              fixedHeader: false,
            }}
            columns={columns}
            data={arr}
            title=""
          />
        </div>
      </div>
    );
  }
}

export default connect(mapStoreToProps, null)(PhotoTable);
