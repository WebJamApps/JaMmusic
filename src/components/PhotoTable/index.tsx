import React, { Dispatch } from 'react';
import MUIDataTable, { MUIDataTableColumnDef } from 'mui-datatables';
import HtmlReactParser from 'html-react-parser';
import type { AnyAction } from 'redux';
import type { Iimage } from '../../redux/mapStoreToProps';
import type { MusicDashboardController } from '../../containers/MusicDashboard/MusicDashboardController';

const setColumns = (): MUIDataTableColumnDef[] => {
  const columns: MUIDataTableColumnDef[] = [];
  const titles = ['Thumbnail', 'Title', 'Caption', 'Link', 'Type', 'Modify'];
  for (let i = 0; i < titles.length; i += 1) {
    const label = titles[i];// eslint-disable-line security/detect-object-injection
    columns.push({
      name: label.toLowerCase(),
      label,
      options: {
        filter: false,
        sort: true,
        customBodyRender: (value:any) => {
          if (typeof value !== 'string' && label !== 'Modify') value = '';
          return (
          <div style={{
            margin: 0, fontSize: '12pt', maxWidth: '200px',
          }}
          >
            {label !== 'Modify' ? HtmlReactParser(value) : value}
          </div>
          );
        },
      },
    });
  }
  return columns;
};

type PTprops = {
  dispatch: Dispatch<AnyAction>,
  auth: { token: string },
  images: Iimage[],
  controller:MusicDashboardController,
};

export const PhotoTable = (props: PTprops):JSX.Element => {

  const editPic = (data: Iimage): boolean => {
    const { dispatch } = props;
    delete data.modify;
    dispatch({ type: 'EDIT_PIC', data });
    return true;
  };

  const addThumbs = (arr: Iimage[]): Iimage[] => {
    const { controller } = props;
    const newArr = arr;/* eslint-disable security/detect-object-injection */
    for (let i = 0; i < arr.length; i += 1) { // eslint-disable-next-line security/detect-object-injection
      newArr[i].thumbnail = `<img src=${arr[i].url} width="200px"/>`;
      const deletePicId = `deletePic${newArr[i]._id}`;// eslint-disable-line security/detect-object-injection
      const editPicId = `editPic${newArr[i]._id}`;// eslint-disable-line security/detect-object-injection
      newArr[i].link = `<a href=${arr[i].url} target="_blank">click to view</a>`;// eslint-disable-line security/detect-object-injection
      newArr[i].caption = newArr[i].comments === 'showCaption' ? 'display' : 'hide';
      newArr[i].modify = (// eslint-disable-line security/detect-object-injection
        <div>
          <button
            type="button"
            id={deletePicId}
            onClick={() => {
              controller.deleteData('deleteImage', newArr[i]._id);
            }}
          >
            Delete Pic
          </button>
          <p>{' '}</p>
          <button type="button" id={editPicId} onClick={() => { editPic(newArr[i]); }}>
            Edit Pic
          </button>
        </div>
      );
    }
    return newArr;
  };

  let arr: Iimage[] = props.images;
  arr = addThumbs(arr);
  return (
      <div className="photoTable">
        <div style={{ maxWidth: '9in', margin: 'auto' }}>
          <MUIDataTable
            options={{
              filterType: 'dropdown',
              pagination: false,
              responsive: 'standard',
              filter: false,
              download: false,
              search: false,
              print: false,
              viewColumns: false,
              selectableRows: 'none',
              fixedHeader: false,
            }}
            columns={setColumns()}
            data={arr}
            title=""
          />
        </div>
      </div>
  );
};
