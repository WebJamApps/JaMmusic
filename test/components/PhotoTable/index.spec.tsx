/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import renderer from 'react-test-renderer';
import { PhotoTable } from '../../../src/components/PhotoTable';

describe('PhotoTable', () => {
  let photoTable:any, props: { auth: any; images: any; };
  const controller:any = { deleteData: jest.fn() };
  beforeEach(() => {
    props = {
      auth: { token: 'token' },
      images: [{
        _id: '456', url: 'url', title: 'title', type: 'youthPics', comments: 'showCaption',
      }],
    };
    photoTable = renderer.create(<PhotoTable
      auth={props.auth}
      images={props.images}
      dispatch={(fun) => fun}
      controller={controller}
    />);
  });
  it('renders correctly', () => { expect(photoTable.toJSON()).toMatchSnapshot(); });
});
