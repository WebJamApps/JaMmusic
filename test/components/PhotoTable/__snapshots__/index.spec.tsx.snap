// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoTable renders correctly 1`] = `
<div
  className="photoTable"
>
  <div
    style={
      Object {
        "margin": "auto",
        "maxWidth": "9in",
      }
    }
  >
    <div
      className="MuiPaper-root MUIDataTable-paper-2 undefined MuiPaper-elevation4 MuiPaper-rounded"
    >
      <div
        className="MUIDataTableFilterList-root-13"
      />
      <div
        className="MUIDataTable-responsiveBase-5"
        style={
          Object {
            "height": "auto",
            "position": "relative",
          }
        }
      >
        <table
          className="MuiTable-root MUIDataTable-tableRoot-4"
          role="grid"
          tabIndex="0"
        >
          <caption
            className="MUIDataTable-caption-11"
          >
            
          </caption>
          <thead
            className="MuiTableHead-root MUIDataTableHead-main-15"
            role={null}
          >
            <tr
              className="MuiTableRow-root MUIDataTableHeadRow-root-19 MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-31"
                data-colindex={0}
                data-tableid="03084344512877164"
                onMouseDown={[Function]}
                scope="col"
              >
                <span
                  className="MUIDataTableHeadCell-contentWrapper-41"
                >
                  <button
                    aria-describedby={null}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MUIDataTableHeadCell-toolButton-40"
                    data-testid="headcol-0"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    title="Sort"
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <div
                        className="MUIDataTableHeadCell-sortAction-36"
                      >
                        <div
                          className="MUIDataTableHeadCell-data-35"
                        >
                          Thumbnail
                        </div>
                        <div
                          className="MUIDataTableHeadCell-sortAction-36"
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-38"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={-1}
                          />
                        </div>
                      </div>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-31"
                data-colindex={1}
                data-tableid="03084344512877164"
                onMouseDown={[Function]}
                scope="col"
              >
                <span
                  className="MUIDataTableHeadCell-contentWrapper-41"
                >
                  <button
                    aria-describedby={null}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MUIDataTableHeadCell-toolButton-40"
                    data-testid="headcol-1"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    title="Sort"
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <div
                        className="MUIDataTableHeadCell-sortAction-36"
                      >
                        <div
                          className="MUIDataTableHeadCell-data-35"
                        >
                          Title
                        </div>
                        <div
                          className="MUIDataTableHeadCell-sortAction-36"
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-38"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={-1}
                          />
                        </div>
                      </div>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-31"
                data-colindex={2}
                data-tableid="03084344512877164"
                onMouseDown={[Function]}
                scope="col"
              >
                <span
                  className="MUIDataTableHeadCell-contentWrapper-41"
                >
                  <button
                    aria-describedby={null}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MUIDataTableHeadCell-toolButton-40"
                    data-testid="headcol-2"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    title="Sort"
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <div
                        className="MUIDataTableHeadCell-sortAction-36"
                      >
                        <div
                          className="MUIDataTableHeadCell-data-35"
                        >
                          Caption
                        </div>
                        <div
                          className="MUIDataTableHeadCell-sortAction-36"
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-38"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={-1}
                          />
                        </div>
                      </div>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-31"
                data-colindex={3}
                data-tableid="03084344512877164"
                onMouseDown={[Function]}
                scope="col"
              >
                <span
                  className="MUIDataTableHeadCell-contentWrapper-41"
                >
                  <button
                    aria-describedby={null}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MUIDataTableHeadCell-toolButton-40"
                    data-testid="headcol-3"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    title="Sort"
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <div
                        className="MUIDataTableHeadCell-sortAction-36"
                      >
                        <div
                          className="MUIDataTableHeadCell-data-35"
                        >
                          Link
                        </div>
                        <div
                          className="MUIDataTableHeadCell-sortAction-36"
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-38"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={-1}
                          />
                        </div>
                      </div>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-31"
                data-colindex={4}
                data-tableid="03084344512877164"
                onMouseDown={[Function]}
                scope="col"
              >
                <span
                  className="MUIDataTableHeadCell-contentWrapper-41"
                >
                  <button
                    aria-describedby={null}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MUIDataTableHeadCell-toolButton-40"
                    data-testid="headcol-4"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    title="Sort"
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <div
                        className="MUIDataTableHeadCell-sortAction-36"
                      >
                        <div
                          className="MUIDataTableHeadCell-data-35"
                        >
                          Type
                        </div>
                        <div
                          className="MUIDataTableHeadCell-sortAction-36"
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-38"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={-1}
                          />
                        </div>
                      </div>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-31"
                data-colindex={5}
                data-tableid="03084344512877164"
                onMouseDown={[Function]}
                scope="col"
              >
                <span
                  className="MUIDataTableHeadCell-contentWrapper-41"
                >
                  <button
                    aria-describedby={null}
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MUIDataTableHeadCell-toolButton-40"
                    data-testid="headcol-5"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    title="Sort"
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <div
                        className="MUIDataTableHeadCell-sortAction-36"
                      >
                        <div
                          className="MUIDataTableHeadCell-data-35"
                        >
                          Modify
                        </div>
                        <div
                          className="MUIDataTableHeadCell-sortAction-36"
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-38"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={-1}
                          />
                        </div>
                      </div>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MUIDataTableBodyRow-root-48 undefined MuiTableRow-hover"
              data-testid="MUIDataTableBodyRow-0"
              id="MUIDataTableBodyRow-03084344512877164-0"
              onClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-52"
                data-colindex={0}
                data-tableid="03084344512877164"
                data-testid="MuiDataTableBodyCell-0-0"
              >
                <div
                  className="MUIDataTableBodyCell-root-52"
                >
                  <div
                    style={
                      Object {
                        "fontSize": "12pt",
                        "margin": 0,
                        "maxWidth": "200px",
                      }
                    }
                  >
                    <img
                      src="url"
                      width="200px"
                    />
                  </div>
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-52"
                data-colindex={1}
                data-tableid="03084344512877164"
                data-testid="MuiDataTableBodyCell-1-0"
              >
                <div
                  className="MUIDataTableBodyCell-root-52"
                >
                  <div
                    style={
                      Object {
                        "fontSize": "12pt",
                        "margin": 0,
                        "maxWidth": "200px",
                      }
                    }
                  >
                    title
                  </div>
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-52"
                data-colindex={2}
                data-tableid="03084344512877164"
                data-testid="MuiDataTableBodyCell-2-0"
              >
                <div
                  className="MUIDataTableBodyCell-root-52"
                >
                  <div
                    style={
                      Object {
                        "fontSize": "12pt",
                        "margin": 0,
                        "maxWidth": "200px",
                      }
                    }
                  >
                    display
                  </div>
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-52"
                data-colindex={3}
                data-tableid="03084344512877164"
                data-testid="MuiDataTableBodyCell-3-0"
              >
                <div
                  className="MUIDataTableBodyCell-root-52"
                >
                  <div
                    style={
                      Object {
                        "fontSize": "12pt",
                        "margin": 0,
                        "maxWidth": "200px",
                      }
                    }
                  >
                    <a
                      href="url"
                      target="_blank"
                    >
                      click to view
                    </a>
                  </div>
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-52"
                data-colindex={4}
                data-tableid="03084344512877164"
                data-testid="MuiDataTableBodyCell-4-0"
              >
                <div
                  className="MUIDataTableBodyCell-root-52"
                >
                  <div
                    style={
                      Object {
                        "fontSize": "12pt",
                        "margin": 0,
                        "maxWidth": "200px",
                      }
                    }
                  >
                    youthPics
                  </div>
                </div>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-52"
                data-colindex={5}
                data-tableid="03084344512877164"
                data-testid="MuiDataTableBodyCell-5-0"
              >
                <div
                  className="MUIDataTableBodyCell-root-52"
                >
                  <div
                    style={
                      Object {
                        "fontSize": "12pt",
                        "margin": 0,
                        "maxWidth": "200px",
                      }
                    }
                  >
                    <div>
                      <button
                        id="deletePic456"
                        onClick={[Function]}
                        type="button"
                      >
                        Delete Pic
                      </button>
                      <p>
                         
                      </p>
                      <button
                        id="editPic456"
                        onClick={[Function]}
                        type="button"
                      >
                        Edit Pic
                      </button>
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        aria-live="polite"
        className="MUIDataTable-liveAnnounce-12"
      />
    </div>
  </div>
</div>
`;
